src/cpu/aarch64/matmul/brgemm_matmul_copy_utils.cpp

```diff
@@ -0,0 +1,2625 @@
+    ZReg vmm_zero = z31;
+
+    ZReg get_comp_acc(int i) { return ZReg(comp_acc_idx_ - i); }
+    ZReg get_vmm_zp_comp_res(int i) { return get_comp_acc(i); }
+    ZReg get_vmm_oscale_comp_res(int i) { return ZReg(i); }
+
+    inline void vmovdqa64(ZReg vmm, const void *addr) {
+        assert(!"under construction");
+#if 0
+        mov(imm_addr64, reinterpret_cast<size_t>(addr));
+        jit_generator::vmovdqa64(vmm, ptr[imm_addr64]);
+#endif
+    }
+
+    inline ZReg get_vmm(int blk, int idx) {
+        if (idx < 0 || idx >= 32) assert(!"idx > vregs");
+        assert(IMPLICATION(!is_ymm_, idx < blk_sz_ && blk >= 0));
+        auto reg_idx = blk_sz_ * blk + idx;
+        return ZReg(reg_idx);
+    }
+    inline void load(int blk, int i, bool is_tail) {}
+    inline void kmovq(Xbyak_aarch64::PReg k, size_t q) {
+        assert(!"under construction");
+#if 0
+    mov(regq_tmp, q);
+    jit_generator::kmovq(k, regq_tmp);
+#endif
+    }
+    virtual void init_permute() {}
+    virtual void copy_4x64(int nrows, int ncolumns) {}
+    inline void dot_product(ZReg v1, ZReg v2, ZReg v3) {
+        assert(!"under construction");
+#if 0
+            vpdpbusd(v1, v2, v3,
+                    mayiuse(avx512_core) ? EvexEncoding : VexEncoding);
+#endif
+    }
+    void generate() override;
+};
+
+template <>
+inline void jit_brgemm_matmul_copy_b_int8_t<sve_512>::load(
+        int blk, int i, bool is_tail) {
+    assert(!"under construction");
+#if 0
+    auto vmm_src = get_vmm(blk, i % k_blk_step_);
+    auto src_load = is_tail ? vmm_src | kTail | T_z : vmm_src;
+    vmovdqu8(src_load, EVEX_compress_addr(reg_src, i * src_stride_));
+#endif
+}
+
+template struct jit_brgemm_matmul_copy_b_int8_t<sve_512>;
+template struct jit_brgemm_matmul_copy_b_int8_t<sve_256>;
+
+struct jit_sve_512_core_brgemm_matmul_copy_b_int8_t
+    : public jit_brgemm_matmul_copy_b_int8_t<sve_512> {
+
+    jit_sve_512_core_brgemm_matmul_copy_b_int8_t(
+            const brgemm_matmul_conf_t *conf)
+        : jit_brgemm_matmul_copy_b_int8_t<sve_512>(conf) {}
+
+private:
+    void init_permute() override {
+        assert(!"under construction");
+#if 0
+        alignas(64) static constexpr const int64_t idx_lo_256[8]
+                = {0, 1, 2, 3, 8, 9, 10, 11};
+        alignas(64) static constexpr const int64_t idx_hi_256[8]
+                = {4, 5, 6, 7, 12, 13, 14, 15};
+
+        alignas(64) static constexpr const int64_t idx_lo_128[8]
+                = {0, 1, 8, 9, 4, 5, 12, 13};
+        alignas(64) static constexpr const int64_t idx_hi_128[8]
+                = {2, 3, 10, 11, 6, 7, 14, 15};
+
+        vmovdqa64(vreg_idx_lo_256, (const void *)idx_lo_256);
+        vmovdqa64(vreg_idx_hi_256, (const void *)idx_hi_256);
+        vmovdqa64(vreg_idx_lo_128, (const void *)idx_lo_128);
+        vmovdqa64(vreg_idx_hi_128, (const void *)idx_hi_128);
+#endif
+    }
+
+    void copy_4x64(int nrows, int ncolumns) override {
+        assert(!"under construction");
+#if 0
```

Comment:
I see a lot of `#if 0` all over the code. What is the purpose of this code? Do you plan to move it to Aarch64 instructions later? If no, I would recommend to remove such code.